pkg/sqlite/load.go

```diff
@@ -1001,3 +1001,57 @@ func (s *sqlLoader) DeprecateBundle(path string) error {
 
 	return tx.Commit()
 }
+
+func (s *sqlLoader) RemoveStrandedBundles() ([]string, error) {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		tx.Rollback()
+	}()
+
+	bundles, err := s.rmStrandedBundles(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return bundles, tx.Commit()
+}
+
+func (s *sqlLoader) rmStrandedBundles(tx *sql.Tx) ([]string, error) {
+	strandedBundles := make([]string, 0)
+
+	strandedBundleQuery := `SELECT name FROM operatorbundle WHERE name NOT IN (select operatorbundle_name from channel_entry)`
+	rows, err := tx.QueryContext(context.TODO(), strandedBundleQuery)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var name sql.NullString
+	for rows.Next() {
+		if err := rows.Scan(&name); err != nil {
+			return nil, err
+		}
+		if name.Valid {
+			strandedBundles = append(strandedBundles, fmt.Sprintf(`"%s"`, name.String))
+		}
+	}
+
+	if len(strandedBundles) == 0 {
+		return nil, nil
+	}
+
```

Comment:
doesn't `rows` need to be closed before this transaction (right now it's deferred to the end)?

Reply:
ah, yep, making a quick change